sklearn/pipeline.py

```diff
@@ -481,7 +481,7 @@ def _inverse_transform(self, X):
         return Xt
 
     @if_delegate_has_method(delegate='_final_estimator')
-    def score(self, X, y=None):
+    def score(self, X, y=None, **score_params):
```

Comment:
What I'm saying is that, despite the PR description referring to `fit_params`, this behaves quite differently. `fit_params` is routed to each step according to `__`-delimited namespaces. By introducing `**score_params` that are all routed to the score function only, we are promising that we will never do anything more sophisticated with any named arguments to score. I'm not convinced we have a use case that motivates such a general implementation, when we could just accept `sample_weight=None`.